staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/metrics/metrics_test.go

```diff
@@ -0,0 +1,174 @@
+}
+
+func (t *testEnvelopeService) Status(ctx context.Context) (*kmsservice.StatusResponse, error) {
+	if t.disabled {
+		return nil, fmt.Errorf("Envelope service was disabled")
+	}
+	return &kmsservice.StatusResponse{KeyID: t.keyVersion}, nil
+}
+
+func (t *testEnvelopeService) SetDisabledStatus(status bool) {
+	t.disabled = status
+}
+
+func (t *testEnvelopeService) SetAnnotations(annotations map[string][]byte) {
+	t.annotations = annotations
+}
+
+func (t *testEnvelopeService) Rotate() {
+	i, _ := strconv.Atoi(t.keyVersion)
+	t.keyVersion = strconv.FormatInt(int64(i+1), 10)
+}
+
+func newTestEnvelopeService() *testEnvelopeService {
+	return &testEnvelopeService{
+		keyVersion: testKeyVersion,
+	}
+}
+
+func TestEnvelopeMetrics(t *testing.T) {
+
+	envelopeService := newTestEnvelopeService()
+	envelopeTransformer := kmstypes.NewEnvelopeTransformer(envelopeService, "providerName",
+		func(ctx context.Context) (string, error) {
+			return "", nil
+		}, aestransformer.NewGCMTransformer)
+
+	dataCtx := value.DefaultContext([]byte(testContextText))
+
+	okTransformer := value.PrefixTransformer{Prefix: []byte("k8s:enc:kms:v2:"), Transformer: envelopeTransformer}
+
+	testCases := []struct {
+		desc    string
+		prefix  value.Transformer
+		metrics []string
+		want    string
+	}{
+		{
+			desc:   "ok keyID total",
+			prefix: value.NewPrefixTransformers(nil, okTransformer),
+			metrics: []string{
+				"apiserver_envelope_encryption_keyid_total",
+			},
+			want: fmt.Sprintf(`
+				# HELP apiserver_envelope_encryption_keyid_total [ALPHA] Number of times a keyID is used split by transformation type and provider.
+				# TYPE apiserver_envelope_encryption_keyid_total counter
+				apiserver_envelope_encryption_keyid_total{key_id="%s",observer_type="encrypted_object_keyid",provider_name="providerName",transformation_type="from_storage"} 1
+        		apiserver_envelope_encryption_keyid_total{key_id="%s",observer_type="plugin_returned_keyid",provider_name="providerName",transformation_type="to_storage"} 1
+        		apiserver_envelope_encryption_keyid_total{key_id="%s",observer_type="last_status_keyid",provider_name="providerName",transformation_type="from_storage"} 1
+        		apiserver_envelope_encryption_keyid_total{key_id="%s",observer_type="last_status_keyid",provider_name="providerName",transformation_type="to_storage"} 1
+				`, testKeyHash, testKeyHash, testEmptyKeyHash, testEmptyKeyHash),
+		},
+		{
+			desc:   "ok cache fill percent",
+			prefix: value.NewPrefixTransformers(nil, okTransformer),
+			metrics: []string{
+				"apiserver_envelope_encryption_dek_cache_fill_percent",
+			},
+			want: `
+				# HELP apiserver_envelope_encryption_dek_cache_fill_percent [ALPHA] Percent of the cache slots currently occupied by cached DEKs.
+				# TYPE apiserver_envelope_encryption_dek_cache_fill_percent gauge
+				apiserver_envelope_encryption_dek_cache_fill_percent 0
```

Comment:
We should test that a percentage is recorded, since you seem to fix that in this PR.